src/mame/irem/olibochu.cpp

```diff
@@ -1,201 +1,220 @@
-          OBC5B
-          OBC4B
- SW1 Z80  OBC3B
-          OBC2B
- SW2      OBC1B   18.432MHz
+        * Verify the HC55536 clock, is it from the master
+          clock or an R/C oscillator on the sound board?
 
+        * Is the "CHU HAS FOOD" sample ever played? Nothing
+          is written to sound_command when a Chu eats a Boo
+          dropping.
 
-VIDEO M-47B-A:
-
-    OBC10      OBC11     OBC9
-    OBC12
-              2125     C-3          2114
-              2125                  2114
-     2125     2125
-     2125     2125
-     2125
-     2125              C-2          2114
-                                    2114
-               OBC14     OBC13
-
-***************************************************************************/
+        * This is the log of the output latch:
+            - Cold boot: Q0-6 off
+            - POST: Q1 on
+            - after POST: Q4-5 on
+            - Title Screen: Q4 is disabled
+            - Stage/Intermission Start: Q5 off/on
+            - Intermission: Q6 on
+            - after Intermission: Q6 off
+            - Warm boot: Q1/Q5 on
+          Are one of the inputs Z80 /WAIT at least?
 
+***********************************************************************/
 #include "emu.h"
 
 #include "cpu/z80/z80.h"
 #include "machine/clock.h"
 #include "machine/gen_latch.h"
+#include "machine/output_latch.h"
 #include "machine/timer.h"
 #include "sound/ay8910.h"
 #include "sound/hc55516.h"
+#include "video/resnet.h"
 
 #include "emupal.h"
 #include "screen.h"
 #include "speaker.h"
 #include "tilemap.h"
 
+
 namespace {
 
 class olibochu_state : public driver_device
 {
 public:
-	olibochu_state(const machine_config &mconfig, device_type type, const char *tag) :
+	olibochu_state(machine_config const &mconfig, device_type type, char const *tag) :
 		driver_device(mconfig, type, tag),
 		m_maincpu(*this, "maincpu"),
+		m_outlatch(*this, "outlatch"),
 		m_audiocpu(*this, "audiocpu"),
+		m_screen(*this, "screen"),
+		m_timer(*this, "timer"),
 		m_gfxdecode(*this, "gfxdecode"),
 		m_palette(*this, "palette"),
 		m_videoram(*this, "videoram"),
 		m_colorram(*this, "colorram"),
-		m_spriteram(*this, "spriteram%u", 0),
-		m_soundlatch(*this, "soundlatch%u", 0),
+		m_spriteram(*this, "spriteram%u", 0U),
+		m_soundlatch(*this, "soundlatch%u", 0U),
+		m_ay8910(*this, "ay8910"),
 		m_cvsd(*this, "cvsd"),
+		m_cvsd_clock(*this, "cvsd_clock") ,
 		m_samplerom(*this, "samples")
 	{ }
 
 	void olibochu(machine_config &config);
 
+	// port handlers
 	DECLARE_INPUT_CHANGED_MEMBER(palette_changed) { adjust_palette(); }
 
 protected:
+	// initialization
 	virtual void machine_start() override;
 	virtual void machine_reset() override;
 	virtual void video_start() override;
 
 private:
+	// constants
+	static inline XTAL constexpr MASTER_CLOCK = 18.432_MHz_XTAL;
+	static inline u16 constexpr HTOTAL = 384, HBSTART = 256, HBEND = 0; // see TODO
+	static inline u16 constexpr VTOTAL = 256, VBSTART = 248, VBEND = 8;
```

Comment:
You say probable raw screen params, I don't think VTOTAL of 256 is likely. If I wanted to take a guess (and I rather not want to blindly guess raw screen params): 18342000/3/(262*384)